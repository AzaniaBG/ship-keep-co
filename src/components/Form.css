.form {
    display: flex;
    flex-direction: column;
}
.form-input {
    margin-bottom: 1em;
}
.form-input input {
    width: 20em;
    padding: .5em;
}
.invalid {
    color: red;
}
.form-input.invalid {
    border-color: red;
}