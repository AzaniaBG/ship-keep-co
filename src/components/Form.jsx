import { useState } from 'react';
import { UseInputState } from '../hooks/UseInputState';

export default function Form() {

    const [firstName, setFirstName] = UseInputState("");
    
    return (
        <form>
            <label>Name</label>
            <input 
                type="text" 
                value={firstName}
                onChange={(e) => setFirstName(e.target.value)}
            />
        </form>

    );
}
