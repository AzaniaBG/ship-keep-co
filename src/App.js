import Form from './components/Form';
import './App.css';

function App() {
  return (
    <div>
      <header>
      
      </header>
      <Form />
    </div>
  );
}

export default App;
