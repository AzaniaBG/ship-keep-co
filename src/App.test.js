import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Form from './components/Form';
import App from './App';

test.skip('renders form component', () => {
  render(<App />);
  const component = renderer.create(<Form />).toJSON();
  expect(component).toMatchSnapshot();
});
// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
