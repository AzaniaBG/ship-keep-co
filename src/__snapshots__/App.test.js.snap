// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form component 1`] = `
<form>
  <label>
    Name
  </label>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    data-testid="input"
    placeholder="name"
    type="text"
  />
</form>
`;
